src/java.desktop/share/classes/javax/swing/plaf/metal/MetalBorders.java

```diff
@@ -241,41 +263,90 @@ public void paintBorder(Component c, Graphics g, int x, int y,
                 shadow = MetalLookAndFeel.getControlInfo();
             }
 
-              g.setColor(background);
-              // Draw outermost lines
-              g.drawLine( 1, 0, w-2, 0);
-              g.drawLine( 0, 1, 0, h-2);
-              g.drawLine( w-1, 1, w-1, h-2);
-              g.drawLine( 1, h-1, w-2, h-1);
+            Graphics2D g2d = (Graphics2D) g;
+            AffineTransform at = g2d.getTransform();
+            Stroke oldStk = g2d.getStroke();
+            Color oldColor = g2d.getColor();
+            boolean resetTransform;
+            int stkWidth = 1;
+
+            // if m01 or m10 is non-zero, then there is a rotation or shear
+            // skip resetting the transform
+            resetTransform = (at.getShearX() == 0) && (at.getShearY() == 0);
+
+            if (resetTransform) {
```

Comment:
Please clarify why we should reset the current transform? How it will work if the user sets a transform and render this to the BufferedImage.

Reply:
On fractional scaling, painting of lines and borders have issues (lines seem to be separated or look uneven) as shown above, https://github.com/openjdk/jdk/pull/10274#issue-1373421530. In order to paint them uniformly and next to each other, the idea is to render at 1x scaling, apply the required thickness/scaling to the lines at this point (when transform is reset) and then the original (old) transform is restored at the end. 

This idea was referenced from a similar issue https://github.com/openjdk/jdk/pull/7449#issuecomment-1068218648.